rewards/service.go

```diff
@@ -2,28 +2,135 @@ package rewards
+	"github.com/rs/zerolog"
 )
 
+// PayoutStatus - current state of the payout status
+type PayoutStatus struct {
+	Unverified string `json:"unverified" valid:"in(off|processing|complete)"`
+	Uphold     string `json:"uphold" valid:"in(off|processing|complete)"`
+	Gemini     string `json:"gemini" valid:"in(off|processing|complete)"`
+	Bitflyer   string `json:"bitflyer" valid:"in(off|processing|complete)"`
+}
+
+// S3GetObjectAPI - interface to allow for a GetObject mock
+type S3GetObjectAPI interface {
+	GetObject(ctx context.Context, params *s3.GetObjectInput, optFns ...func(*s3.Options)) (*s3.GetObjectOutput, error)
+}
+
+func extractPayoutStatus(ctx context.Context, client S3GetObjectAPI, bucket, object string) (*PayoutStatus, error) {
+	logger := logging.Logger(ctx, "rewards.extractPayoutStatus")
+	// get the object with the client
+	out, err := client.GetObject(
+		ctx, &s3.GetObjectInput{
+			Bucket: &bucket,
+			Key:    &object,
+		})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get payout status: %w", err)
+	}
+	defer func() {
+		err := out.Body.Close()
+		logger.Error().Err(err).Msg("failed to close s3 result body")
+	}()
+	var payoutStatus = new(PayoutStatus)
+
+	// parse body json
+	if err := inputs.DecodeAndValidateReader(ctx, payoutStatus, out.Body); err != nil {
+		return nil, payoutStatus.HandleErrors(err)
+	}
+
+	return payoutStatus, nil
+}
+
+// HandleErrors - handle any errors in input
+func (ps *PayoutStatus) HandleErrors(err error) *handlers.AppError {
+	return handlers.ValidationError("invalid payout status", err)
+}
+
+// Decode - implement decodable
+func (ps *PayoutStatus) Decode(ctx context.Context, input []byte) error {
+	return json.Unmarshal(input, ps)
+}
+
+// Validate - implement validatable
+func (ps *PayoutStatus) Validate(ctx context.Context) error {
+	isValid, err := govalidator.ValidateStruct(ps)
+	if err != nil {
+		return err
+	}
+	if !isValid {
+		return errors.New("invalid input")
+	}
+	return nil
+}
+
+type appLogger struct {
+	*zerolog.Logger
+}
+
+// Logf - implement smithy-go/logging.Logger
+func (al *appLogger) Logf(classification awslogging.Classification, format string, v ...interface{}) {
+	al.Debug().Msg(fmt.Sprintf(format, v...))
+}
+
 // NewService - create a new rewards service structure
-func NewService(ctx context.Context, ratio ratios.Client) *Service {
-	return &Service{
-		jobs:   []srv.Job{},
-		ratios: ratio,
+func NewService(ctx context.Context, ratio ratios.Client, s3client S3GetObjectAPI) (*Service, error) {
+	logger := logging.Logger(ctx, "rewards.NewService")
+
+	logger.Info().Msg("getting aws region")
+	// get the aws region from ctx
+	region, ok := ctx.Value(appctx.AWSRegionCTXKey).(string)
+	if !ok || len(region) == 0 {
+		region = "us-west-2"
 	}
+
+	logger.Info().Msg("checking s3 client")
+	if s3client == nil {
+		logger.Info().Str("region", region).Msg("setting up s3 client")
+		// aws config
+		cfg, err := config.LoadDefaultConfig(
+			context.TODO(),
```

Comment:
why context.TODO() here instead of the context passed in?

Reply:
https://github.com/brave-intl/bat-go/pull/1482

copy paste oversight, the above corrects this.